testsuite/module_verify.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Aptos
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+import shutil
+import subprocess
+
+from verify_core.common import query_backup_latest_version, clear_artifacts
+
+
+def main():
+    # collect all required ENV variables
+    REQUIRED_ENVS = [
+        "BUCKET",
+        "SUB_DIR",
+        "BACKUP_CONFIG_TEMPLATE_PATH",
+    ]
+    if not all(env in os.environ for env in REQUIRED_ENVS):
+        raise Exception("Missing required ENV variables")
+
+    BACKUP_CONFIG_TEMPLATE_PATH = os.environ["BACKUP_CONFIG_TEMPLATE_PATH"]
+
+    if not os.path.exists(BACKUP_CONFIG_TEMPLATE_PATH):
+        raise Exception("BACKUP_CONFIG_TEMPLATE_PATH does not exist")
+    with open(BACKUP_CONFIG_TEMPLATE_PATH, "r") as f:
+        config = f.read()
+        if "aws" in config and shutil.which("aws") is None:
+            raise Exception("Missing required AWS CLI for pulling backup data from S3")
+
+    if os.environ.get("REUSE_BACKUP_ARTIFACTS", "true") == "true":
+        print("[main process] clearing existing backup artifacts")
+        clear_artifacts()
+    else:
+        print("[main process] skipping clearing backup artifacts")
+
+    LATEST_VERSION = query_backup_latest_version(BACKUP_CONFIG_TEMPLATE_PATH)
+
+    print(f"Latest version: {LATEST_VERSION}")
+
+    # run verify-modules
+    os.mkdir("local")
```

Comment:
Is there a way to borrow the forge shell and filesystem abstractions?

then we can maek this easy to test

Reply:
yea definitely, though there's some further cleanup that needs to happen. probably in a separate PR :p